src/Rings/MPolyQuo.jl

```diff
@@ -947,3 +947,59 @@ function dim(a::MPolyQuoIdeal)
   return a.dim
 end
 #######################################################
+
+
+################################################################################
+#
+#  Minimalizing a set of subalgebra generators in graded case
+#
+################################################################################
+@doc Markdown.doc"""
+    minimal_subalgebra_generators(V::Vector{S}) where S <: Union{MPolyElem{U}, MPolyQuoElem{U}} where U
+
+Given a vector `V` collecting homogeneous elements of a graded multivariate polynomial ring or quotient ring, say `R`, return a minimal subset of these elements which still generate the subalgebra of `R` generated by all elements collected in `V`.
+
+NOTE: The conditions on `V` and `R` are automatically checked.
+
+# Examples
+```jldoctest
+julia> R, (x,y,z) =  GradedPolynomialRing(QQ, ["x", "y", "z"])
+(Multivariate Polynomial Ring in x, y, z over Rational Field graded by
+  x -> [1]
+  y -> [1]
+  z -> [1], MPolyElem_dec{fmpq, fmpq_mpoly}[x, y, z])
+
+julia> V = [x,y,x^2+y^2]
+3-element Vector{MPolyElem_dec{fmpq, fmpq_mpoly}}:
+ x
+ y
+ x^2 + y^2
+
+julia> minimal_subalgebra_generators(V)
+2-element Vector{MPolyElem_dec{fmpq, fmpq_mpoly}}:
+ y
+ x
+```
+"""
+function minimal_subalgebra_generators(V::Vector{S}) where S <: Union{MPolyElem{U}, MPolyQuoElem{U}} where U
```

Comment:
```suggestion
function minimal_subalgebra_generators(V::Vector{S}) where S <: Union{MPolyElem, MPolyQuoElem}
```
Unless I am mistaken, there is no need to qualify `U` here (also in the docstring, and in a bunch of other places)

Reply:
Compre also with subalgebra_membership, please do whatever you believe is fine